### PR TITLE
Add parameter mappers to OperationSpec

### PR DESCRIPTION
**Pull request by daschult**
Resolves https://github.com/Azure/autorest.typescript/issues/94.

### CODE REVIEW

File: src/vanilla/ClientModelExtensions.cs [hunk 1]
```diff
@@ -630,314 +671,246 @@ public static string ConstructMapper(this IModelType type, string serializedName
...
-            {
-                builder.AppendLine($"xmlName: '{xmlName}',");
-            }
-
-            if (!isCaseSensitive)
+            value.Object(mapper =>
             {
-                serializedName = serializedName.ToLower();
-            }
+                bool wroteXmlName = isXML && !string.IsNullOrEmpty(xmlName) && xmlName != serializedName;
```
Comment (RikkiGibson) [inline src/vanilla/ClientModelExtensions.cs]: should this be called `shouldWriteXmlName` or similar?

File: test/azure/generated/AzureResource/models/mappers.ts [hunk 2]
```diff
@@ -14,205 +14,179 @@ export const CloudError = CloudErrorMapper;
 export const BaseResource = BaseResourceMapper;
 
 export const ErrorModel = {
-  required: false,
-  serializedName: 'Error',
+  serializedName: "Error",
```
Comment (RikkiGibson) [inline test/azure/generated/AzureResource/models/mappers.ts]: does the linter prefer double quotes? I seem to recall our ms-rest-js and autorest.typescript linters having contradictory settings.

File: src/vanilla/ClientModelExtensions.cs (hunk 1 above)
Comment (daschult) [reply src/vanilla/ClientModelExtensions.cs]: Changed to `addXmlNameParameterValue`. This variable really keeps track of whether we should add the xmlName parameter that is passed to this function or whether we should use the xmlName property on the property variable.

File: test/azure/generated/AzureResource/models/mappers.ts (hunk 2 above)
Comment (daschult) [reply test/azure/generated/AzureResource/models/mappers.ts]: That sounds familiar, but I'm just going to go with whatever settings we use in the ms-rest-js repository. If we start getting linter errors in autorest.typescript, then we should update the autorest.typescript linter rules to be the same as ms-rest-js.

**Comment by RikkiGibson** [comment]
The bundle size check is failing due to "stdout maxbuffer exceeded". Will fiddle with that in another PR, probably.

**Pull request merged**